pytket_qir/qir.py

```diff
@@ -400,11 +445,80 @@ def condition_zero_block():
                 one=lambda: condition_one_block(),
                 zero=lambda: condition_zero_block(),
             )
+        elif isinstance(op, WASMOp):
+            cregs = _retrieve_registers(circ.bits, BitRegister)
+            args = command.args
+
+            inputs: List[str] = []
+            outputs: List[str] = []
+            for reglist, sizes in [
+                (inputs, op.input_widths),
+                (outputs, op.output_widths),
+            ]:
+                for in_width in sizes:
+                    wasm_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = wasm_bits[0].reg_name
+                    if wasm_bits != list(cregs[regname]):
+                        WASMUnsupportedError("WASM ops must act on entire registers.")
+                    reglist.append(regname)
+
+            reg = circ.get_c_register(inputs[0])
```

Comment:
I don't see where `inputs` has been populated?

Reply:
`reglist, sizes` iterates over the tuples formed by the `(registers, widths)` here:

```
for reglist, sizes in [
                (inputs, op.input_widths),
                (outputs, op.output_widths),
            ]:
``` 

To be honest, this bit of code is brutally nicked from Seyon's code about handling WASM in `pytket.qasm` [here].(https://github.com/CQCL/tket/blob/dc2136fb621b3d1659d5e98c73138564dbb0fa39/pytket/pytket/qasm/qasm.py#L1010)

Reply:
Ah, it's OK, `reglist` is of course a reference to `inputs` and it does get updated.